cmd/kubeadm/app/cmd/upgrade/common.go

```diff
@@ -220,10 +220,7 @@ func enforceRequirements(flags *applyPlanFlags, args []string, dryRun bool, upgr
 	if dupURLScheme {
 		socket := strings.ReplaceAll(cfg.NodeRegistration.CRISocket, kubeadmapiv1.DefaultContainerRuntimeURLScheme+"://", "")
 		cfg.NodeRegistration.CRISocket = kubeadmapiv1.DefaultContainerRuntimeURLScheme + "://" + socket
-		hostname, err := os.Hostname()
-		if err != nil {
-			return nil, nil, nil, errors.Wrapf(err, "failed to get hostname")
-		}
+		hostname := cfg.NodeRegistration.Name
```

Comment:
if cfg.NodeRegistration.Name is empty we should resort to asking os.Hostname?

Reply:
I can implement that.
However looking at the doc for `cfg.NodeRegistration.Name`: https://github.com/kubernetes/kubernetes/blob/32c483ea6ee90a3a81f382563c91034470af8a4a/cmd/kubeadm/app/apis/kubeadm/types.go#L216
```
	// Name is the `.Metadata.Name` field of the Node API object that will be created in this `kubeadm init` or `kubeadm join` operation.
	// This field is also used in the CommonName field of the kubelet's client certificate to the API server.
	// Defaults to the hostname of the node if not provided.
```
I doubt that this field could ever be empty. or am I wrong ?

Reply:
@neolit123 I've reverted to the previous behavior as a fallback:
https://github.com/clementnuss/kubernetes/blob/kubeadm-upgrade-fqdn-fix/cmd/kubeadm/app/cmd/upgrade/common.go#L223-L231

I think with that this PR could be merged. Ideally before the next `1.24` patch release, as this will be a blocker for many users